lib/bundle/cask_dumper.rb

```diff
@@ -8,20 +8,42 @@ def reset!
       @casks = nil
     end
 
+    def casks_data
+      require "cask/caskroom"
+      @casks_data ||= Cask::Caskroom.casks
+    end
+
     def casks
       return [] unless Bundle.cask_installed?
 
       require "cask/caskroom"
-
       @casks ||= Cask::Caskroom.casks.map(&:to_s)
-      @casks.map { |cask| cask.chomp " (!)" }
-            .uniq
+    end
+
+    def casks_config
+      return {} unless Bundle.cask_installed?
+
+      @casks_config ||= Hash[
```

Comment:
How might I add a default value to this hash while preserving the memoization behaviour?

Reply:
Feels like it's worth populating the entire hash rather than setting a default value here. Use `||= begin` to cache an entire block.